src/ImageManipulation.php

```diff
@@ -1057,4 +1062,98 @@ protected function castDimension($value, $dimension)
         }
         return $value;
     }
+
+    private function copyImageBackend()
+    {
+        // Store result in new DBFile instance
+        /** @var DBFile $file */
+        $file = DBField::create_field(
+            'DBFile',
+            [
+                'Filename' => $this->getFilename(),
+                'Hash' => $this->getHash(),
+                'Variant' => $this->getVariant()
+            ]
+        );
+
+        $file->setImageBackend($this->getImageBackend());
+        return $file->setOriginal($this);
+    }
+
+    public function setAttribute($name, $value)
+    {
+        $file = $this->copyImageBackend();
+        $file->setMutableAttributes(array_merge(
+            $this->attributes,
+            [$name => $value]
+        ));
+
+        return $file;
+    }
+
+    public function setMutableAttributes($attributes)
+    {
+        $this->attributes = $attributes;
+    }
+
+    protected function getDefaultAttributes(): array
+    {
+        if (!$this->getIsImage()) {
```

Comment:
I'm assuming that Images will always have a Height and a Width here.

Reply:
This method is on ImageManipulation, it's probably fair to assume that we're dealing with an Image at this point?

Reply:
`ImageManipulation` is applied to file ... which is very confusing. All those methods are available on `File`, they just return null in most cases.